### PR TITLE
Adjust km distance (latitude-longitude) calculation

### PR DESCRIPTION
**Pull request by ADBond**
Includes:
* fixing an error with calculation of distance from latitude/longitude (a typo in placement of parentheses)
  * changing in fact to a simpler and quicker method for performing this calculation
* separating out the sql expression-function for this from the comparison-level it was in, to allow it to be recomposed in further comparison situations
* unit tests for this calculation

Was originally attempting to work on some comparisons between array columns (including distances), but thought this made more sense as an individual and separate PR.

### CODE REVIEW

**Comment by ADBond** [comment]
Re: the details of the calculation, I benchmarked three forms of calculation (the original (but corrected) version with `atan`, the alternative `asin` version, and the now-in-the-branch `acos` version (from the spherical law of cosines)).

This was in the case of literal numeric expressions, from a single row table, and a multi-row table (script below).
bm_3

File: splink/misc.py [hunk 1]
```diff
@@ -141,6 +141,28 @@ def calculate_reduction_ratio(N, cartesian):
...
+        cast(
+            acos(
+                {partial_distance_sql}
+            ) * {EARTH_RADIUS_KM}
+            as float
+        )
+    """
+    return distance_km_sql
+
+
```
Comment (ThomasHepworth) [inline splink/misc.py:165]: I'd plop this into its own script.

`comparison_level_sql.py`?

**Review by ThomasHepworth — APPROVED** [review]
Looks good - can you just migrate the sql out of misc and into its own script then merge.

I'll make a new release once that's done.

File: splink/misc.py (hunk 1 above)
Comment (ADBond) [reply splink/misc.py:165]: Sure thing, that makes sense - wasn't overly convinced that this was the most sensible place to put this.

**Comment by ADBond** [comment]
@ThomasHepworth Thanks for your note, have implemented.
I noticed after also that the new test was not passing in the workflow (specifically a test expecting 0 wasn't sufficiently close on this system being run on). I have now added a second route to passing, of being within `1e-3` of the correct answer, i.e. corresponding to 1 metre, which seems sufficient imo.
I'll leave this unmerged to allow a chance for any objections, but happy if you wish to go ahead, or I can address anything else tomorrow.

**Comment by RobinL** [comment]
Could you also make sure the linting passes - on the tests, feel free to make liberal use of `# noqa: 501` for line length stuff (see e.g. [here](https://github.com/moj-analytical-services/splink/blob/e42feb84ee6b88903b94137a9e23c3b11b8248d9/tests/test_columns_selected.py#L105) )

**Review by RobinL — APPROVED** [review]

**Pull request merged**